supervisor/addons/validate.py

```diff
@@ -336,7 +359,15 @@ def _migrate(config: dict[str, Any]):
         vol.Optional(ATTR_DEVICES): [str],
         vol.Optional(ATTR_UDEV, default=False): vol.Boolean(),
         vol.Optional(ATTR_TMPFS, default=False): vol.Boolean(),
-        vol.Optional(ATTR_MAP, default=list): [vol.Match(RE_VOLUME)],
+        vol.Optional(ATTR_MAP, default=list): [
+            vol.Schema(
+                {
+                    vol.Required("type"): str,
+                    vol.Optional("read_only"): bool,
+                    vol.Optional("path"): str,
```

Comment:
```suggestion
                    vol.Optional(ATTR_PATH): str,
```
As mentioned above we prefer constants when the value is reused. `ATTR_PATH` doesn't currently exist in `supervisor.const` or `supervisor.addons.const` so it'll have to be defined.

And simply `str` for `path` seems a bit loose. At a minimum it has to start with `/` since we want absolute paths not relative ones. Also multiple of these cannot have the same value for path, that would need be validated in a separate validator function. And I'm not sure we should allow choosing absolutely any path within the container. @pvizeli and @agners are there any paths we don't want to allow here?

Reply:
That's a great point. I did some local testing with several conditions and the containers fail to start. The error messages from Docker make sense in each case:

![image](https://github.com/home-assistant/supervisor/assets/9720393/122c7e35-623c-493f-ab8c-c89764cba6db)
![image](https://github.com/home-assistant/supervisor/assets/9720393/5d8ab09a-d186-4b52-9831-946e24dc9b67)
![image](https://github.com/home-assistant/supervisor/assets/9720393/ed42bd05-6b3d-475e-b5b9-73a028661943)

I'm not sure what Supervisor best practices are, should these conditions also be validated at the config level so as to catch these issues earlier? Or is it sufficient that these are handled when trying to start the Addon?

Reply:
I don't seem to come up with a good reason to filter certain paths. I mean you can do silly things, but it will affect your add-on only, so :man_shrugging:

Reply:
> I'm not sure what Supervisor best practices are, should these conditions also be validated at the config level so as to catch these issues earlier? Or is it sufficient that these are handled when trying to start the Addon?

I think I would not try to "reimplement" these checks on config levels. An add-on developer certainly starts his add-on once before deploying :sweat_smile: 

Not sure if we can make the Docker errors a bit nicer, e.g. strip the API/http related part. But that is something which bubbles up also in other places, and might be worth addressing separately.

Reply:
We can just make an validation function to load the string as Path (and return also a Path) - but also check if that is an absolute Path. Pathib have 2 Path object - virtual or physic - we just can use the virtual version to validate

Reply:
Hi! Sorry @pvizeli, I was out of town on vacation last week. That sounds like a good plan! I can add the path validation logic. I should be able to get to that tomorrow.

Any thoughts on the discussion above?
https://github.com/home-assistant/supervisor/pull/4694#discussion_r1399723085